c7n/commands.py

```diff
@@ -122,5 +152,102 @@ def logs(options, policies):
             e['message'])
 
 
+def schema(options):
+    """
+    Output information about the resources, actions and filters available.
+    """
+    if options.json:
+        schema_json_dump(options.resource)
+        return
+        
+    load_resources()
+    resource_mapping = resource_vocabulary()
+
+    if options.summary:
+        schema_summary(resource_mapping)
+        return
+
+    # Here are the formats for what we accept:
+    # - No argument
+    #   - List all available RESOURCES
+    # - RESOURCE
+    #   - List all available actions and filters for supplied RESOURCE
+    # - RESOURCE.actions
+    #   - List all available actions for supplied RESOURCE
+    # - RESOURCE.actions.ACTION
+    #   - Show class doc string and schema for supplied action
+    # - RESOURCE.filters
+    #   - List all available filters for supplied RESOURCE
+    # - RESOURCE.filters.FILTER
+    #   - Show class doc string and schema for supplied filter
+
+    if not options.resource:
+        resource_list = {'resources': sorted(resources.keys()) }
+        print(yaml.safe_dump(resource_list, default_flow_style=False))
+        return
+
+    # Format is RESOURCE.CATEGORY.ITEM
+    components = options.resource.split('.')
+
+    #
+    # Handle resource
+    #
+    resource = components[0].lower()
+    if resource not in resource_mapping:
+        print('{} is not a valid resource'.format(resource))
+        sys.exit(2)
+
+    if len(components) == 1:
+        del(resource_mapping[resource]['docs'])
+        output = {resource: resource_mapping[resource]}
+        print(yaml.safe_dump(output))
+        return
+
+    #
+    # Handle category
+    #
+    category = components[1].lower()
+    if category not in ('actions', 'filters'):
+        print("Valid choices are 'actions' and 'filters'.  You supplied '{}'".format(category))
+        sys.exit(2)
+    
+    if len(components) == 2:
+        output = "No {} available for resource {}.".format(category, resource)
+        if category in resource_mapping[resource]:
+            output = {resource: {category: resource_mapping[resource][category]}}
+        print(yaml.safe_dump(output))
+        return
+
+    #
+    # Handle item
+    #
+    item = components[2].lower()
+    if item not in resource_mapping[resource][category]:
+        print('{} is not in the {} list for resource {}'.format(item, category, resource))
+        sys.exit(2)
+
+    if len(components) == 3:
+        docstring = resource_mapping[resource]['docs'][category][item]
```

Comment:
I did a `pprint` with indent=2 for the schema.  Here are two examples.

```
$ custodian schema ec2.filters.onhour

Help:
-----

Filter instances by state.

Try to simplify construction for policy authors by automatically
filtering elements (filters or actions) to the instances states
they are valid for.

For more details see http://goo.gl/TZH9Q5

Schema:
-------

{   'additionalProperties': False,
    'properties': {   'debug': {   'type': 'boolean'},
                      'default_tz': {   'type': 'string'},
                      'onhour': {   'maximum': 24,
                                    'minimum': 0,
                                    'type': 'integer'},
                      'opt-out': {   'type': 'boolean'},
                      'tag': {   'type': 'string'},
                      'type': {   'enum': ['onhour']},
                      'weekends': {   'type': 'boolean'},
                      'weekends-only': {   'type': 'boolean'}},
    'required': ['onhour', 'default_tz', 'type'],
    'type': 'object'}
```

```
$ custodian schema ec2.actions.terminate

Help:
-----

Terminate a set of instances.

While ec2 offers a bulk delete api, any given instance can be configured
with api deletion termination protection, so we can't use the bulk call
reliabily, we need to process the instances individually. Additionally
If we're configured with 'force' then we'll turn off instance termination
protection.

Schema:
-------

{   'additionalProperties': False,
    'properties': {   'force': {   'type': 'boolean'},
                      'type': {   'enum': ['terminate']}},
    'required': ['type'],
    'type': 'object'}
```

Look ok?